crates/shark-error/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+use std::io::{self, Write};
+
+use source::SourcePosition;
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+pub mod source;
+
+pub struct SharkError<'a> {
+    pub kind: SharkErrorKind,
+    pub start_position: &'a SourcePosition<'a>,
+    pub end_position: &'a SourcePosition<'a>,
```

Comment:
I should migrate this back to Range<SourcePosition<'a>>. The only reason why this currently isn't is because I thought it caused a bug